mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb

```diff
@@ -47,28 +47,45 @@ func resourceComputeRouterInterface() *schema.Resource {
-				ConflictsWith:    []string{"interconnect_attachment"},
 				Optional:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
-				AtLeastOneOf:     []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
-				Description:      `The name or resource link to the VPN tunnel this interface will be linked to. Changing this forces a new interface to be created. Only one of vpn_tunnel and interconnect_attachment can be specified.`,
+				AtLeastOneOf:     []string{"ip_range", "interconnect_attachment", "subnetwork", "vpn_tunnel"},
+				ConflictsWith:    []string{"interconnect_attachment", "subnetwork"},
+				Description:      `The name or resource link to the VPN tunnel this interface will be linked to. Changing this forces a new interface to be created. Only one of vpn_tunnel, interconnect_attachment or subnetwork can be specified.`,
 			},
 			"interconnect_attachment": {
 				Type:             schema.TypeString,
-				ConflictsWith:    []string{"vpn_tunnel"},
 				Optional:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
-				AtLeastOneOf:     []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
-				Description:      `The name or resource link to the VLAN interconnect for this interface. Changing this forces a new interface to be created. Only one of vpn_tunnel and interconnect_attachment can be specified.`,
+				AtLeastOneOf:     []string{"ip_range", "interconnect_attachment", "subnetwork", "vpn_tunnel"},
+				ConflictsWith:    []string{"subnetwork", "vpn_tunnel"},
+				Description:      `The name or resource link to the VLAN interconnect for this interface. Changing this forces a new interface to be created. Only one of interconnect_attachment, subnetwork or vpn_tunnel can be specified.`,
 			},
 			"ip_range": {
+				Type:           schema.TypeString,
+				Optional:       true,
+				ForceNew:       true,
+				Computed:       true,
+				AtLeastOneOf:   []string{"ip_range", "interconnect_attachment", "subnetwork", "vpn_tunnel"},
+				Description:    `The IP address and range of the interface. The IP range must be in the RFC3927 link-local IP space. Changing this forces a new interface to be created.`,
+			},
+			"private_ip_address": {
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				AtLeastOneOf: []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
-				Description:  `IP address and range of the interface. The IP range must be in the RFC3927 link-local IP space. Changing this forces a new interface to be created.`,
+				Description:  `The regional private internal IP address that is used to establish BGP sessions to a VM instance acting as a third-party Router Appliance. Changing this forces a new interface to be created.`,
+			},
+			"subnetwork": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
```

Comment:
Ah, sorry for the whack-a-mole. I just caught this one when taking a final look. What makes `subnetwork` need to be `Computed`? Will it be set by the API when not sent under some circumstances?

Reply:
Done!